### PR TITLE
Editor Stats - Fix `from_template_selector` property for insert block events.

### LINKED ISSUE #53637: Editor Stats - `from_template_selector` property for block events is always false.
When a pattern is inserted from the page layout selector, the `from_template_selector` property for the block inserted tracks events should be `true`.  However, this seems to be populated as `false`.  Looking at tracks in MC it looks like this has not been tracked a `true` since mid-march. 

FYI @kurt213 since this looks like it has been busted for a couple months.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
#### Changes proposed in this Pull Request
Fixes https://github.com/Automattic/wp-calypso/issues/53637
Updates the filter used to populated `from_template_selector` property on block inserted events to match the string id used in the start page templates component.  

#### Testing instructions

* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Create a new page and select a page layout to insert.
* After the blocks are inserted, verify the `wpcom_block_inserted` events correctly show the `from_template_selector` property as `true`.

### CODE REVIEW

**Comment by Addison-Stavlo** [comment]
I hope to add (either here or in a follow up) an e2e test to catch a future regression here.

**Review by david-szabo97 — DISMISSED** [review]
Nice! Works as expected! 🚢  One small request below.

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 1]
```diff
@@ -285,7 +285,7 @@ const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
 		block_name: name,
 		blocks_replaced: true,
 		// isInsertingPageTemplate filter is set by Starter Page Templates
```
Comment (david-szabo97) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js]: We need to update the comment too.

**Comment by david-szabo97** [comment]
Also, quick question. Is it possible to have the old version of Editing Toolkit on a site? Should we make sure this is backward compatible by checking for both flags?

**Comment by creativecoder** [comment]
> Also, quick question. Is it possible to have the old version of Editing Toolkit on a site? Should we make sure this is backward compatible by checking for both flags?

This is a good question! I think that's a good idea, as we know auto updates don't always work as expected.

File: apps/wpcom-block-editor/src/wpcom/features/tracking.js [hunk 2]
```diff
@@ -285,7 +285,7 @@ const trackInnerBlocksReplacement = ( rootClientId, blocks ) => {
...
-		from_template_selector: applyFilters( 'isInsertingPageTemplate', false ),
+		from_template_selector: applyFilters( 'isInsertingPagePattern', false ),
```
Comment (david-szabo97) [inline apps/wpcom-block-editor/src/wpcom/features/tracking.js]: Let's keep the old hook too for backward compatibility. Let's be on the safe side 😄

Comment (Addison-Stavlo) [reply apps/wpcom-block-editor/src/wpcom/features/tracking.js]: Yeah that makes sense 😁 .  Updated!

**Review by david-szabo97 — APPROVED** [review]

**Pull request merged**